### DISCUSSION TITLE
Does Hono support WebSocket in deno?

### DISCUSSION THREAD

**Discussion by yusukebe** [honojs/hono#755] [Q&A] [answered]
Does Hono support WebSockt in deno?

**Comment by yusukebe** [comment, marked as answer]
Yes, you can write like this:

```ts
app.get('/ws/', (c) => {
  const { response, socket } = Deno.upgradeWebSocket(c.req)
  socket.addEventListener('message', (e) => console.log(e))
  return response
})
```

This is a good article:

* [DenoのWebSocketサーバーの書き方 2022冬編 - Qiita](https://qiita.com/access3151fq/items/7e2bdbd8f24858cf1f49#hono%E3%82%92%E4%BD%BF%E3%81%86%E5%A0%B4%E5%90%88)

**Reply by erf** [reply]
Is there a way to do this in Bun as well ?

**Reply by erf** [reply]
I was able to upgrade to WebSocket using the following:
```
app.get('/ws', (c) => {
  console.log(c)
  const success = c.env.upgrade(c.req.raw)
  return true
})
```
See https://bun.sh/guides/websocket/upgrade

However it was needed to return `true` and not `undefined` as per docs.

Also i would like a way to listen to the various WebSocket messges inside the ws router, and not in the Bun.serve callbacks.
